src/modules/validator.js

```diff
@@ -0,0 +1,96 @@
+class Validator {
+
+    constructor(chluIpfs) {
+        this.chluIpfs = chluIpfs;
+    }
+
+    async validateReviewRecord(reviewRecord, validations = {}) {
+        const rr = cloneReviewRecord(reviewRecord);
+        const v = Object.assign({
+            validateVersion: true,
+            validateMultihash: true,
+            validateHistory: true
+        }, validations);
+        if (v.validateVersion) this.validateVersion(rr);
+        if (v.validateMultihash) await this.validateMultihash(rr, rr.hash.slice(0));
+        if (v.validateHistory) await this.validateHistory(rr);
+    }
+
+    async validateMultihash(obj, expected) {
+        const hashedObj = await this.chluIpfs.reviewRecords.hashReviewRecord(obj);
+        if (expected !== hashedObj.hash) {
+            throw new Error('Mismatching hash');
+        }
+    }
+
+    async validateHistory(reviewRecord) {
+        const history = await this.chluIpfs.reviewRecords.getHistory(reviewRecord);
+        if (history.length > 0) {
+            const opt = {
+                validate: false
+            };
+            const rr = await this.chluIpfs.reviewRecords.readReviewRecord(history[0], opt);
```

Comment:
We call `readReviewRecord` in `getHistory`, are we sure this second call to `readReviewRecord` will come of the local disk or memory and not from another network request?

In either case, I think we can optimise this a bit? Maybe `getHistory` can return an array of review records instead of just multihashes since the records are anyway available while running getHistory.

What do you think @fazo96 ?

Reply:
Doesn't make much difference either way, since if you fetch something from IPFS two times, the first time if it's not cached locally it will use the network. The second time it will 100% only hit the cache. It's really not an expensive operation to do

However it makes sense to return a `{ multihash: reviewRecord }` object for the history

Reply:
Ok I fixed this, but I used this structure `[{multihash, reviewRecord}, ... ]` so that the ordering of the updates gets preserved